src/main/java/com/dislinkt/post/controller/ReactionController.java

```diff
@@ -0,0 +1,94 @@
+package com.dislinkt.post.controller;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.dislinkt.post.dto.ErrorDTO;
+import com.dislinkt.post.dto.ReactionDTO;
+import com.dislinkt.post.dto.ResponseDTO;
+import com.dislinkt.post.service.ReactionService;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/reactions")
+public class ReactionController {
+
+    @Autowired
+    private ReactionService service;
+
+    @GetMapping()
+    public ResponseEntity<ResponseDTO<List<ReactionDTO>>> getAll(){
+        List<ReactionDTO> ret = service.findAll();
+        return new ResponseEntity<>(new ResponseDTO<>(ret), HttpStatus.OK);
+    }
+
+    @GetMapping("/{postId}")
+    public ResponseEntity<ResponseDTO<List<ReactionDTO>>> findByPost(@PathVariable Integer postId){
+        List<ReactionDTO> ret = service.findByPostId(postId);
+        return new ResponseEntity<>(new ResponseDTO<>(ret), HttpStatus.OK);
+    }
+
+    @GetMapping("/likes/{postId}")
+    public ResponseEntity<ResponseDTO<List<ReactionDTO>>> findPostLikes(@PathVariable Integer postId){
+        List<ReactionDTO> ret = service.findAllLikesForPost(postId);
+        return new ResponseEntity<>(new ResponseDTO<>(ret), HttpStatus.OK);
+    }
+
+    @GetMapping("/dislikes/{postId}")
+    public ResponseEntity<ResponseDTO<List<ReactionDTO>>> findPostDislikes(@PathVariable Integer postId){
+        List<ReactionDTO> ret = service.findAllDislikesForPost(postId);
+        return new ResponseEntity<>(new ResponseDTO<>(ret), HttpStatus.OK);
+    }
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity addReaction(@RequestHeader("X-User-Id") UUID personId, @RequestBody ReactionDTO dto){
```

Comment:
Not necessary but would be nice since you'll be calling `@RequestHeader("X-User-Id")` very often if you could try to implement something that would look like this:
`@LoggedUser() UUID persoinId`

Reply:
There you could check if that personId exists so you don't have to check for it in each service. I think that would be very effective. 😄

Reply:
I see what you mean, but honestly using @RequestHeader currently seems like the simpler solution (especially because the name of the header key is the always the same), so I think I will stick with it for now and then maybe change it later once we start doing frontend.